### DISCUSSION TITLE
How to format float in simple notation, not scientific one: e.g. 1695979657.74, not 1.69597965774e+09

### DISCUSSION THREAD

**Discussion by piotr-g** [ohler55/ojg#141] [Q&A] [answered]
The example is a timestamp in seconds since unix epoch time. How can I use json.Number in a struct (unmarshaling and marshaling to float64)?

**Comment by ohler55** [comment]
I'm not clear on the question. Let me try to get some details. Give a struct that contains a json.Number you want the representation in a JSON string to be decimal only format and not scientific notation in the JSON. You also want to marshal and unmarshal the JSON into a float64 presumably into a different struct. Is that correct?

**Reply by piotr-g** [reply]
My main question is how to avoid scientific notation while marshaling float64 to json. I unmarshal one json string to a struct where big float is in a  simple format (without exponent) e.g. 1695979657.74 but while marshaling the same struct to json I'm getting 1.69597965774e+09 in output string. I would prefer to get the number in the same non-scientific format. Second question was about using json.Number (e.g. if sometimes I get json number in quotes). As I understand json.Number is convenient in such situations.

**Reply by ohler55** [reply]
Right now there is not an option on what format is used for floats. It is always 'g' (e.g., fmt.Sprintf("%g", num)). An option could be added to allow a user supplied format directive though. It would be a nice addition I think.

When a number string in a JSON document is encountered an attempt is made to parse the number without a loss of precision. If the number is too large or small then a json.Number is created.

**Reply by ohler55** [reply, marked as answer]
The current develop branch now has a ojg.Option.FloatFormat which can be used when writing floats. Not much I can do about the json.Number other than providing an option for something other than jsonNumber when a number is too large but that doesn't solve the json.Number output flexibility.

**Reply by piotr-g** [reply]
Thanks Peter for a quick answer. That's exactly what I need. BTW any plan for releasing it?

**Reply by ohler55** [reply]
I want to improve the struct support in the ojg/jp package first. So maybe this coming week.

**Reply by ohler55** [reply]
This feature was added in v1.20.0 on Oct 17th.
